app/client/src/components/shared/LocationSearch/index.js

```diff
@@ -500,6 +500,37 @@ function LocationSearch({ route, label }: Props) {
     }
   }
 
+  // Splits the provided text by the searchString in a case insensitive way.
+  function getHighlightParts(text, searchString) {
+    const searchLength = searchString.length;
+    const indices = indicesOf(text, searchString);
+
+    // build an array of the string split up by the searchString that includes
+    // the searchString.
+    const parts = [];
+    let endIndex = 0;
+    let remainder = text;
+    indices.forEach((startIndex) => {
+      // skip if the indices are the same (i.e. results in empty string)
+      if (endIndex !== startIndex) {
+        // add in text up to the start index
+        parts.push(text.substring(endIndex, startIndex));
+      }
+
+      // add in the search part of the text
+      endIndex = startIndex + searchLength;
+      parts.push(text.substring(startIndex, endIndex));
+
+      // keep track of leftover text
+      remainder = text.substring(endIndex);
+    });
+
+    // add in remainder text if applicable
+    if (remainder) parts.push(remainder);
+
+    return parts;
+  }
```

Comment:
@cschwinderg, instead of creating a custom "indicesOf" function that you're only using once here, you could use the `matchAll` method on the source string, passing in the user's search input (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll). That returns a regular expression iterator, which you can spread into an array and map over to get the indices – it yields the same result as the custom "indiciesOf" function you created, but is less code to maintain (notice the "i" case insensitive flag passed to the regular expression too, so no need for toLowerCase() here either).

I'd also not store the search string's length in a variable, as you're only using it once in the block the variable is defined:

```suggestion
function getHighlightParts(text, searchString) {
    const matches = text.matchAll(new RegExp(searchString, 'gi'));
    const indices = [...matches].map((iterator) => iterator.index);

    // build an array of the string split up by the searchString that includes
    // the searchString.
    const parts = [];
    let endIndex = 0;
    let remainder = text;
    indices.forEach((startIndex) => {
      // skip if the indices are the same (i.e. results in empty string)
      if (endIndex !== startIndex) {
        // add in text up to the start index
        parts.push(text.substring(endIndex, startIndex));
      }

      // add in the search part of the text
      endIndex = startIndex + searchString.length;
      parts.push(text.substring(startIndex, endIndex));

      // keep track of leftover text
      remainder = text.substring(endIndex);
    });

    // add in remainder text if applicable
    if (remainder) parts.push(remainder);

    return parts;
  }
```

Reply:
Ah sorry! I was looking at the code first and testing second – I see the issue of the user passing in an opening square bracket, as it causes an "Unterminated Character Class" Regular Expression error...that's a bummer. I know we were trying to escape the user's input before, but you're probably right to just not use a regular expression to parse input after all.

Reply:
Yeah, I wish we could just use Regex like before. There probably is a way to make it work for that particular string, but I'm just afraid that some other input will cause a similar issue. I figure it is easier to create our on function and not have to worry about this.

Reply:
I removed the `searchLength` variable.